### PR TITLE
Fix `Bun.sleepSync` to actually use milliseconds

### LINKED ISSUE #1420: sleepSync takes second and not ms
Hello,

I'm not sure if it's a typing problem or a code problem but the function `sleepSync` seems like to take seconds as parameters but the typing say `Sleep the thread for a given number of milliseconds`.

```javascript
import { sleepSync } from 'bun'

while (true) {
    sleepSync(1)
    console.log(new Date())
}
```
```
2022-09-27T08:10:49.235Z
2022-09-27T08:10:50.235Z
2022-09-27T08:10:51.235Z
2022-09-27T08:10:52.236Z
2022-09-27T08:10:53.236Z
2022-09-27T08:10:54.236Z
```

### PR DESCRIPTION
**Pull request by jwhear**
`Bun.sleepSync` was treating its argument as seconds rather than milliseconds as the docs stated.  This is a breaking change in that the function now behaves as documented.  Fixed relevant tests.

Fixes #1420

### CODE REVIEW

**Comment by Jarred-Sumner** [comment]
Yeah it was stupid to make it take seconds, though I wonder if anyone is using Bun.sleepSync

**Comment by Jarred-Sumner** [comment]
There are a few usages. I think we can still do this breaking change at our stage, but it will be harder to do this after 1.0.

cc @pnxdxt @softwarearchitect817 @johncosta27 @jhmaster2000

image

File: test/bun.js/serve.test.ts [hunk 1]
```diff
@@ -123,15 +123,15 @@ it("request.signal works in leaky case", async () => {
         leaky = req;
         expect(didAbort).toBe(false);
         aborty.abort();
-        await Bun.sleep(2);
+        await Bun.sleep(2000);
```
Comment (Jarred-Sumner) [inline test/bun.js/serve.test.ts]: sleep was already milliseconds

**Review by Jarred-Sumner — APPROVED** [review]
LGTM once changes to tests calling Bun.sleep() are reverted

**Comment by pnodet** [comment]
I would rather have it in `ms` for consistency.

**Comment by xhyrom** [comment]
> LGTM once changes to tests calling Bun.sleep() are reverted

Why do you want to return the tests? They're good now, they use milliseconds.

**Comment by Jarred-Sumner** [comment]
> > LGTM once changes to tests calling Bun.sleep() are reverted
> 
> Why do you want to return the tests? They're good now, they use milliseconds.

They were already in milliseconds.

**Comment by pfgithub** [comment]
Why doesn't this code crash with large numbers?

```zig
const milliseconds = JSValue.fromRef(arguments[0]).asNumber();
if (milliseconds > 0 and std.math.isFinite(milliseconds))
    std.time.sleep(@floatToInt(u64, milliseconds) * std.time.ns_per_ms);
```

For 92233720368547758070000 ms, I would expect floatToInt to crash because "integer part of floating point value out of bounds" and for a smaller values, multiplying by ns_per_ms could overflow

In bun 0.5.7, calling `Bun.sleepSync(9223372036854775807 * 10000)` doesn't crash though. Is bun compiled in ReleaseFast mode or am I missing something?

**Comment by jwhear** [comment]
> For 92233720368547758070000 ms, I would expect floatToInt to crash because "integer part of floating point value out of bounds" and for a smaller values, multiplying by ns_per_ms could overflow
> 
> In bun 0.5.7, calling `Bun.sleepSync(9223372036854775807 * 10000)` doesn't crash though. Is bun compiled in ReleaseFast mode or am I missing something?

In debug builds it panics with `integer part of floating point value out of bounds`.  In production builds it becomes undefined behavior.  We could add a check here to avoid the undefined behavior but arguably trying to sleep for 3 trillion years is pretty close to undefined behavior...

**Comment by Jarred-Sumner** [comment]
> Why doesn't this code crash with large numbers?
> 
> ```zig
> const milliseconds = JSValue.fromRef(arguments[0]).asNumber();
> if (milliseconds > 0 and std.math.isFinite(milliseconds))
>     std.time.sleep(@floatToInt(u64, milliseconds) * std.time.ns_per_ms);
> ```
> 
> For 92233720368547758070000 ms, I would expect floatToInt to crash because "integer part of floating point value out of bounds" and for a smaller values, multiplying by ns_per_ms could overflow
> 
> In bun 0.5.7, calling `Bun.sleepSync(9223372036854775807 * 10000)` doesn't crash though. Is bun compiled in ReleaseFast mode or am I missing something?

We shouldn’t be using .asNumber() in most cases because it is unsafe. .coerce(i32) or .to(i32) clamps it to the desired range 

I think that applies here as well

File: Makefile [hunk 2]
```diff
@@ -569,7 +569,11 @@ c-ares:
 	rm -rf $(BUN_DEPS_DIR)/c-ares/build && \
 	mkdir $(BUN_DEPS_DIR)/c-ares/build && \
 	cd $(BUN_DEPS_DIR)/c-ares/build && \
-    cmake $(CMAKE_FLAGS) -DCMAKE_C_FLAGS="$(CFLAGS) -flto=full" -DCMAKE_BUILD_TYPE=Release -DCARES_STATIC=ON -DCARES_STATIC_PIC=ON -DCARES_SHARED=OFF -G "Ninja" .. && \
+    cmake $(CMAKE_FLAGS) -DCMAKE_C_FLAGS="$(CFLAGS) -flto=full" \
```
Comment (jwhear) [inline Makefile:572]: I don't know why this change is showing up; it's already in main.

Comment (Jarred-Sumner) [reply Makefile:572]: maybe i reverted too much stuff

Comment (jwhear) [reply Makefile:572]: That's probably it.  Then this is me just sneaking that c-ares Makefile change back in.

**Pull request merged**